keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -128,115 +125,66 @@ class SqlResourceRepository(
-          Unknown.name,
-          clock.instant().let(Timestamp::from),
-          instanceIdSupplier.get()
-        )
-        .execute()
     }
   }
 
-  override fun lastKnownState(uid: UID): ResourceStateHistoryEntry =
-    jooq
-      .select(
-        field("state"),
-        field("timestamp")
-      )
-      .from(RESOURCE_STATE)
-      .where(field("uid").eq(uid.toString()))
-      .orderBy(field("timestamp").desc())
-      .limit(1)
-      .fetch()
-      .intoResultSet()
-      .run {
-        if (next()) {
-          ResourceStateHistoryEntry(state, timestamp)
-        } else {
-          throw NoSuchResourceUID(uid)
-        }
-      }
-
-  override fun stateHistory(uid: UID): List<ResourceStateHistoryEntry> =
+  override fun eventHistory(uid: UID): List<ResourceEvent> =
     jooq
       .select(
-        field("state"),
-        field("timestamp")
+        field("json")
       )
-      .from(RESOURCE_STATE)
+      .from(RESOURCE_EVENT)
       .where(field("uid").eq(uid.toString()))
       .orderBy(field("timestamp").desc())
       .fetch()
       .intoResultSet()
       .run {
-        val results = mutableListOf<ResourceStateHistoryEntry>()
+        val results = mutableListOf<ResourceEvent>()
         while (next()) {
-          results.add(ResourceStateHistoryEntry(state, timestamp))
+          results.add(event)
         }
         results
       }
       .apply {
         if (isEmpty()) throw NoSuchResourceUID(uid)
       }
 
-  override fun updateState(uid: UID, state: ResourceState) {
-    // TODO: long term it may make more sense to use 2 tables
-    // one storing the "latest" state and one storing the full
-    // history. Can then do a single tx.
+  override fun appendHistory(event: ResourceEvent) {
     jooq.inTransaction {
-      select(
-        field("state")
-      )
-        .from(RESOURCE_STATE)
-        .where(
-          field("uid").eq(uid.toString())
+      insertInto(RESOURCE_EVENT)
+        .columns(
+          field("uid"),
+          field("timestamp"),
+          field("json")
         )
-        .orderBy(field("timestamp").desc())
-        .limit(1)
-        .forUpdate()
-        .fetch()
-        .intoResultSet()
-        .apply {
-          if (!next() || getString("state") != state.name) {
-            insertInto(RESOURCE_STATE)
-              .columns(
-                field("uid"),
-                field("state"),
-                field("timestamp"),
-                field("instance_id")
-              )
-              .values(
-                uid.toString(),
-                state.name,
-                clock.instant().let(Timestamp::from),
-                instanceIdSupplier.get()
-              )
-              .execute()
-          }
-        }
+        .values(
+          event.uid.toString(),
+          event.timestamp.let(Timestamp::from),
+          objectMapper.writeValueAsString(event)
+        )
+        .execute()
     }
   }
 
   override fun delete(name: ResourceName) {
```

Comment:
Thoughts on a model where we have a deleted flag on the resource but don't remove it from the data store? Was thinking it might make for a better auditing story?

Reply:
Yeah at some point we’ll want to do that. I think for now the key thing is just to be able to get rid of stuff so we can do repeated demos!

Reply:
I'm going to go ahead and merge this. We have #165 to track a proper deletion solution. I agree that keeping audit history and just flagging a resource as inactive is probably the right thing, but I think it's best handled as its own PR.